Lidgren.Network/NetPeer.LatencySimulation.cs

```diff
@@ -197,7 +197,11 @@ internal bool ActuallySendPacket(byte[] data, int numBytes, NetEndPoint target,
 					connectionReset = true;
 					return false;
 				}
-				LogError("Failed to send packet: " + sx);
+				if (sx.SocketErrorCode != SocketError.Success)

```

Comment:
Uh, have you actually confirmed the packets aren't being dropped?

Reply:
I'll check for sure, maybe the packets are dropping somewhere

Reply:
Increasing limits to 25M buffer size didn't solve this issue, It still get this "Success" errors spam.

This values for sysctl:
``` s
net.core.rmem_max=26214400
net.core.rmem_default=26214400
net.core.wmem_max=26214400
net.core.wmem_default=26214400
```

Here is `netstat -s -u` output:
``` s
IcmpMsg:
    InType3: 368354
    InType5: 28
    InType8: 823610
    InType11: 49427
    OutType0: 823610
    OutType3: 41759
    OutType5: 47
Udp:
    30925338821 packets received
    148475 packets to unknown port received
    19988770 packet receive errors
    47150146129 packets sent
    19988770 receive buffer errors
    1072665605 send buffer errors
    IgnoredMulti: 1524
UdpLite:
IpExt:
    OutMcastPkts: 118
    InBcastPkts: 1524
    InOctets: 6256479142490
    OutOctets: 23654397742519
    OutMcastOctets: 4720
    InBcastOctets: 527068
    InNoECTPkts: 32781728773
    InECT1Pkts: 132
    InECT0Pkts: 6578
    InCEPkts: 13
MPTcpExt:
```

Also tried to increase `sendbuffersize` and `receivebuffersize`, but it didn't make any sense.

Reply:
What I mean to say is: do we know for sure that the error really is bogus, or is something going wrong and are we not pulling out the error code correctly?

If there really is an error when sending a packet it should obviously be reported. Why we're getting errors and then failing to find the error code for them is what we need to figure out here.

I took a look at the code in `NetNativeSocket.cs`, and we aren't doing `SetLastError` on the `DllImport` for the Linux signatures. I suspect this is the cause of the errors not being reported and it should be fixed.